packages/devtools_shared/lib/src/deeplink/deeplink_manager.dart

```diff
@@ -23,150 +24,140 @@ class DeeplinkManager {
   /// {"configurations":["Debug","Release","Profile"],"targets":["Runner","RunnerTests"]}
   static final _iosBuildOptionsJsonRegex = RegExp(r'({.*})');
 
+  /// The key to retrieve error message from the returning map of this class's
+  /// APIs.
   static const kErrorField = 'error';
+
+  /// The key to retrieve output json from the returning map of this class's
+  /// APIs.
   static const kOutputJsonField = 'json';
 
+  /// A regex to retrieve the file path from the stdout of iOS or Android
+  /// analyzers.
+  ///
+  /// Example stdout:
+  ///
+  /// result saved in /path/to/json/file.json
   static final _outputFilePathRegex = RegExp(r'result saved in (.*.json)');
 
   @visibleForTesting
   Future<ProcessResult> runProcess(
     String executable, {
     required List<String> arguments,
-    required String workingDirectory,
   }) {
     return Process.run(
       executable,
       arguments,
-      workingDirectory: workingDirectory,
     );
   }
 
-  Future<ProcessResult> _runFlutterCommand(List<String> arguments) {
-    final String flutterRoot = Platform.environment['FLUTTER_ROOT']!;
-    return runProcess(
-      'flutter',
-      arguments: arguments,
-      workingDirectory: flutterRoot,
+  @visibleForTesting
+  String? getFlutterBinary() {
+    final String? flutterRoot = Platform.environment['FLUTTER_ROOT'];
+    if (flutterRoot == null) {
+      return null;
+    }
+    return path.join(
+      flutterRoot,
+      'bin',
+      Platform.isWindows ? 'flutter.bat' : 'flutter',
     );
   }
 
-  Future<Map<String, String?>> _runFlutterCommandAndReadOutputFile(
-    List<String> arguments,
-  ) async {
-    final ProcessResult result = await _runFlutterCommand(arguments);
+  Future<String> _runFlutterCommand(
+    List<String> arguments, {
+    required RegExp outputMatcher,
+  }) async {
+    final flutterPath = getFlutterBinary();
+    if (flutterPath == null) {
+      throw 'Running Flutter command is not supported in this devtool version';
+    }
+    final result = await runProcess(flutterPath, arguments: arguments);
     if (result.exitCode != 0) {
-      return <String, String>{
-        kErrorField:
-            'Flutter command exit with non-zero error code ${result.exitCode}\n${result.stderr}',
-      };
+      throw 'Flutter command exit with non-zero error code ${result.exitCode}\n${result.stderr}';
```

Comment:
can we throw a specific type of exception instead of throwing a string? e.g. maybe ProcessException here and FormatException below. Here and elsewhere where we are throwing a String